_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+Special thanks to Prof. Joan Bruna and his class at NYU, [Mathematics of Deep Learning](https://github.com/joanbruna/MathsDL-spring19), and to Cinjon Resnick, who introduced me to DFL and helped me putting up this guide.
+
+Thank you to Avital Oliver, Matt Johnson, Dougal MacClaurin, David Duvenaud, and Ricky Chen for useful contributions to this guide.
+
+Thank you to Tinghao Li, Chandra Prakash Konkimalla, Manikanta Srikar Yellapragada, Shan-Conrad Wolf, Deshana Desai, Yi Tang, Zhonghui Hu for helping me prepare the notes.
+
+Finally, thank you to all my fellow students who attended the recitations and provided valuable feedback.
+
+<div class="deps-graph">
+<iframe class="deps" src="/assets/nodes-deps.svg" width="200"></iframe>
+<div>Concepts used in Neural ODEs. Click to navigate.</div>
+</div>
+
+# Why
+
+Neural ODEs are neural network models which generalize standard layer to layer propagation to continuous depth models. Starting from the observation that the forward propagation in neural networks is equivalent to one step of discretation of an ODE, we can construct and efficiently train models by via ODEs. On top of providing a novel family of architectures, notably for invertible density models and continuous time series, neural ODEs also provide a memory efficiency gain in supervised learning tasks.
+
+In this curriculum, we will go through all the background topics necessary to understand these models. At the end, you should be able to implement neural ODEs and apply them to different tasks.
+
+<br />
+
+# Common resources:
+
+1. (Süli & Mayers): [An Introduction to Numerical Analysis](https://www.cambridge.org/core/books/an-introduction-to-numerical-analysis/FD8BCAD7FE68002E2179DFF68B8B7237#)
+2. (Quarteroni et al.): [Numerical Mathematics](https://www.springer.com/us/book/9783540346586?token=holiday18&utm_campaign=3_fjp8312_us_dsa_springer_holiday18&gclid=Cj0KCQiAvebhBRD5ARIsAIQUmnlViB7VsUn-2tABSAhIvYaJgSEqmJXD7F4A7EgyDQtY9v_GeUsNif8aArGAEALw_wcB)
+
+# 1 Numerical solution of ODEs - Part 1
+  **Motivation**: ODEs are used to mathematically model a number of natural processes and phenomena. The study of their numerical 
+    simulations is one of the main topics in numerical analysis and of fundamental importance in applied sciences. To understand Neural ODEs, we need to first understand how ODEs are solved with numerical techniques.
+
+  **Topics**:
+
+  1. Initial values problems
+  2. One-step methods
+  3. Consistency and convergence
+
+  **Required Reading**:
+
+  1. Sections 12.1-4 from (Süli & Mayers)
+  2. Sections 11.1-3 from (Quarteroni et al.)
+  
+  **Notes**: Here is a [link](/assets/nodes_notes/week1.pdf) to our notes for the lesson.
+  
+  **Optional Reading**:
+
+  1. Runge-Kutta methods: Section 12.5 from (Süli & Mayers)
+  2. [Prof. Trefethen's class ODEs and Nonlinear Dynamics 4.2](http://podcasts.ox.ac.uk/odes-and-nonlinear-dynamics-42)
+
+  **Questions**:
+
+  1. Exercise 1 in Section 11.12 of (Quarteroni et al.)
```

Comment:
Exercise 1 solution:
	- What’s up with E_1 = O(h * * 3)? Is that supposed to be **?
	- The link doesn’t work.
	- Is this a complete solution? Seems more like a hint.

Reply:
expanded

Reply:
- "On the other hand we can bound E_1 as E_1 = O(h^2)" --> "On the other hand, E_1 is bounded above by O(h^2)".
- The link isn't showing up right.

Reply:
fixed